### ISSUE TITLE
BUG - Image Stays On Screen When Deleting Last Image In Directory

### ISSUE THREAD

**Issue by RainfallWithin** [d2phap/ImageGlass#837]
### System information:
 - Windows OS version: Windows 10 1903
 - ImageGlass version: 7.6.4.30

### To Reproduce
1. Delete the last image in a directory.
2. 
3. 
4. 

### Actual behavior:
ImageGlass continues to show the image that was deleted.

### Expected behavior:
ImageGlass should show a blank screen with the message 'Reached the last image'.

### Screenshots:
Before last image deletion:
![Screenshot (688)](https://user-images.githubusercontent.com/69644154/90165229-a74f9700-dd98-11ea-8d82-af58aa5f7597.png)

After last image deletion:
![Screenshot (689)](https://user-images.githubusercontent.com/69644154/90165299-bdf5ee00-dd98-11ea-96d1-bc65c2fb30ec.png)

### Additional context:
Previous versions of ImageGlass correctly showed a blank screen after deleting the last image. ImageGlass Moon 7.6.7.12 still has the same bug.

**Comment by RainfallWithin** [comment]
I have installed ImageGlass 7.0.7.26 and will continue using that version for the time being.

Here are screenshots of the expected behaviour from ImageGlass 7.0.7.26.

Before last image deletion:

![Screenshot (694)](https://user-images.githubusercontent.com/69644154/90166846-1e862a80-dd9b-11ea-8a7a-7eebeefdfb2e.png)

After last image deletion:

![Screenshot (695)](https://user-images.githubusercontent.com/69644154/90166882-2e057380-dd9b-11ea-99fd-24e2ba510ee1.png)

**Comment by d2phap** [comment]
Hi @RainfallWithin 

Just asking if you checked the option: "Loop back viewer to the first image". 
The "Reached the last image" message only shown if this option is unchecked.

![image](https://user-images.githubusercontent.com/3154213/90207152-1cae7c80-de18-11ea-8ca3-e82a8847bf51.png)

**Comment by RainfallWithin** [comment]
Hello d2phap,

Yes, I always have that option unchecked. I configure ImageGlass this way every time I install it. In ImageGlass 7.0.7.26 the screen becomes blank when deleting the last image, but in versions after that the deleted image stays on screen.

Here are my settings:
![{76F6C4AA-ECD0-4437-BE1E-F1A3524A74E8}](https://user-images.githubusercontent.com/69644154/90207894-bc074b80-dde7-11ea-9530-6c745ce6f271.png)

![{E25BB4EE-3DB7-4917-92DF-323B6182FC23}](https://user-images.githubusercontent.com/69644154/90207898-bf023c00-dde7-11ea-80a4-2a3fc88a0c49.png)

**Comment by d2phap** [comment]
Thanks @RainfallWithin 
I will investigate on it

**Comment by RainfallWithin** [comment]
Thank you very much, d2phap. I will be patient. I'm happy to spend time testing anything you suggest.

**Comment by fire-eggs** [comment]
I took a look at this, and it seems a little difficult to fix this "cleanly". Here are my findings for Phap to make the executive decision ...

The change is to frmMain.cs\NextPic(), currently at line 626-631. The code currently reads:
```
            if (!Configs.IsSlideshow && !Configs.IsLoopBackViewer) {
                //Reach end of list
                if (tempIndex >= Local.ImageList.Length) {
                    ShowToastMsg(Configs.Language.Items[$"{Name}._LastItemOfList"], 1000);
                    return;
                }
```
So in the "normal" case, when the end of the list is reached (loop back is OFF), then the last image is still displayed and the Toast message is on top.

My change, which provides the specific behavior requested by the OP, is:
```
            if (!Configs.IsSlideshow && !Configs.IsLoopBackViewer) {
                //Reach end of list
                if (tempIndex >= Local.ImageList.Length) {

                    // Issue #837 : when the last image in the list has been deleted,
                    // and IsLoopBackViewer is off, do not keep showing the (now deleted) image.
                    if (Local.CurrentIndex >= Local.ImageList.Length) {
                        picMain.Image = null;
                    }

                    ShowToastMsg(Configs.Language.Items[$"{Name}._LastItemOfList"], 1000);
                    return;
                }
```
With this code, in the "last image deleted" case, the image is cleared and the Toast message appears. The "normal" case is unchanged.

I say this isn't "clean" because the title bar still shows the details about the now deleted image, including inaccurate file counts (nn/nn). If I throw an `UpdateStatusBar()` call in there, then the title bar is blanked except for even worse file counts (e.g. image 13/12).

I think the "clean" solution may be to call `NextPic(-1)` instead of `picMain.Image = null`. Now the "normal" case is still unchanged; for the "last image deleted" case, the display switches to the new "last" image, with the Toast message on top, and the title bar showing valid info. But doing so does not provided the solution as requested by the OP.

**Comment by RainfallWithin** [comment]
@fire-eggs, I really appreciate the time you have spent on this.

The change of code you have suggested is exactly what I was hoping for, and your worry about the title bar still showing the details of the now deleted last image are not a problem to me, and in fact, might be useful in some circumstances.

I often sort and delete hundreds of images that are very similar in appearance. A blank screen when deleting the last image is very helpful. If the screen isn't blank after I've deleted the last image, I don't notice that I've reached the end and try to continue the sorting.

> I say this isn't "clean" because the title bar still shows the details about the now deleted image

I tested ImageGlass 7.0.7.26 that had the preferred behaviour (showing a blank screen after deleting the last image), and its behaviour matches what you described.

![{86127EB9-5AD2-4172-8CF2-22CE5E69A314}](https://user-images.githubusercontent.com/69644154/92292220-59076100-ef1c-11ea-94fa-d6faa5682bf6.png)

![Screenshot (853)](https://user-images.githubusercontent.com/69644154/92292224-5dcc1500-ef1c-11ea-84c6-a59cfcbf99b9.png)

![Screenshot (855)](https://user-images.githubusercontent.com/69644154/92292226-60c70580-ef1c-11ea-82b0-74581b8d30bd.png)

As you can see, the last image's name, 'Screenshot 620.png', still shows in the title bar after deletion.

This is something I never noticed until you pointed it out, and in my opinion is not an issue, and may in fact be helpful if I want to quickly check the name of the last image I deleted.

> ..., including inaccurate file counts (nn/nn). If I throw an UpdateStatusBar() call in there, then the title bar is blanked except for even worse file counts (e.g. image 13/12).

In ImageGlass 7.0.7.26, you can see from my screenshots that the file count stays at 2/2 when deleting the last image, even thought there is 1 image left in the directory. I think is is completely OK to leave it this way. Techically I'm still on the last image when the screen has become blank, and the first thing myself and other users will do after reaching the end is to close ImageGlass, or select another image which would update the file count.

**Comment by fire-eggs** [comment]
Thanks for the feedback! I should have confirmed the behavior of an older version.

**Comment by d2phap** [comment]
Should be fixed in [ImageGlass 9.0.9.1230](https://github.com/d2phap/ImageGlass/releases/tag/9.0.9.1230)
